src/app/admin/pontuacao/form-pontuacao/form-pontuacao.component.ts

```diff
@@ -32,23 +32,39 @@ export class FormPontuacaoComponent implements OnInit, OnChanges {
 
   ngOnChanges(changes: SimpleChanges) {
     // tslint:disable-next-line:max-line-length
-    if ('atividade' in changes && changes.atividade.currentValue) { // Se trocou a atividade, seta os campos correspondentes
-      this.tituloAtividade = changes.atividade.currentValue.titulo;
-      this.formPontuacao.controls['atividade'].setValue(changes.atividade.currentValue.id);
+    if ('atividade' in changes && !changes.atividade.firstChange) { // Se trocou a atividade, seta os campos correspondentes
+      if (changes.atividade.currentValue) {
+        this.tituloAtividade = changes.atividade.currentValue.titulo;
+        this.formPontuacao.controls['atividade'].setValue(changes.atividade.currentValue.id);
+      } else {
+        this.tituloAtividade = '';
+        this.formPontuacao.controls['atividade'].reset();
+      }
     }
-    if (changes.aluno && changes.aluno.currentValue) { // Se trocou o aluno, seta os campos correspondentes
-      this.nomeAluno = changes.aluno.currentValue.fullName;
-      this.formPontuacao.controls['aluno'].setValue(changes.aluno.currentValue.id);
+    if (changes.aluno && !changes.aluno.firstChange) { // Se trocou o aluno, seta os campos correspondentes
```

Comment:
aqui e em todos os ifs você tem que fazer a checagem assim: 
```suggestion
    if ('aluno' in changes && !changes.aluno.firstChange) { // Se trocou o aluno, seta os campos correspondentes
```
ou ainda, 
```suggestion
    if (typeof changes['aluno'] !== "undefined" && !changes.aluno.firstChange) { // Se trocou o aluno, seta os campos correspondentes
```